src/ripple/core/SQLInterface_global.h

```diff
@@ -0,0 +1,203 @@
+    std::string const& dbname);
+
+void
+loadManifest(
+    soci::session& session,
+    std::string const& dbTable,
+    ManifestCache& mCache,
+    beast::Journal j);
+
+void
+saveManifest(
+    soci::session& session,
+    std::string const& dbTable,
+    std::function<bool(PublicKey const&)> isTrusted,
+    hash_map<PublicKey, Manifest> const& map,
+    beast::Journal j);
+
+void
+addValidatorManifest(soci::session& session, std::string const& serialized);
+
+std::pair<PublicKey, SecretKey>
+loadNodeIdentity(soci::session& session);
+
+std::unordered_set<PeerReservation, beast::uhash<>, KeyEqual>
+loadPeerReservationTable(
+    soci::session& session,
+    beast::Journal j);
+
+void
+insertPeerReservation(
+    soci::session& session,
+    PublicKey const& nodeId,
+    std::string const& description);
+
+void
+deletePeerReservation(soci::session& session, PublicKey const& nodeId);
+
+/* Shard asquire DB */
+
+std::unique_ptr<DatabaseCon>
+makeAcquireDB(
+    DatabaseCon::Setup const& setup,
+    DatabaseCon::CheckpointerSetup const& checkpointerSetup);
+
+void
+insertAcquireDBIndex(soci::session& session, std::uint32_t index_);
+
+std::pair<bool, std::optional<std::string>>
+selectAcquireDBLedgerSeqs(soci::session& session, std::uint32_t index);
+
+std::tuple<bool, std::optional<std::string>, std::optional<std::string>>
+selectAcquireDBLedgerSeqsHash(soci::session& session, std::uint32_t index);
+
+void
+updateAcquireDB(
+    soci::session& session,
+    std::shared_ptr<Ledger const> const& ledger,
+    std::uint32_t index,
+    std::uint32_t lastSeq,
+    std::optional<std::string> const& seqs);
+
+/* Archive DB */
+
+std::unique_ptr<DatabaseCon>
+makeArchiveDB(boost::filesystem::path const& dir, std::string const& dbName);
+
+void
+readArchiveDB(
+    soci::session& session,
+    std::function<void(std::string const&, int)> const& func);
+
+void
+insertArchiveDB(
+    soci::session& session,
+    LedgerIndex shardIndex,
+    std::string const& url);
+
+void
+deleteFromArchiveDB(soci::session& session, LedgerIndex shardIndex);
+
+void
+dropArchiveDB(soci::session& session);
+
+/* State DB */
+
+struct SavedState
+{
+    std::string writableDb;
+    std::string archiveDb;
+    LedgerIndex lastRotated;
+};
+
+void
+initStateDB(
+    soci::session& session,
+    BasicConfig const& config,
+    std::string const& dbName);
+
+LedgerIndex
+getCanDelete(soci::session& session);
+
+LedgerIndex
+setCanDelete(soci::session& session, LedgerIndex canDelete);
+
+SavedState
+getSavedState(soci::session& session);
+
+void
+setSavedState(soci::session& session, SavedState const& state);
+
+void
+setLastRotated(soci::session& session, LedgerIndex seq);
+
+/* DatabaseBody DB */
+
+std::pair<std::unique_ptr<DatabaseCon>, std::optional<std::uint64_t>>
+openDatabaseBodyDb(DatabaseCon::Setup const& setup, boost::filesystem::path path);
+
+std::uint64_t
+databaseBodyDoPut(
+    soci::session& session,
+    std::string&& data,
+    std::string&& path,
```

Comment:
I don't understand why these were changed to be rvalue refs. Do not appear to ever be moved from. What's the motivation for the change?

Reply:
Done. Replaced by const&.